File: src/main/java/com/github/miyashy/karatesample/service/UserService.java
package com.github.miyashy.karatesample.service;

import com.github.miyashy.karatesample.domain.User;
import com.github.miyashy.karatesample.domain.repository.UserRepository;
import java.util.Optional;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.UUID;

@RequiredArgsConstructor
@Service
public class UserService {
    private final UserRepository userRepository;

    public User add(String name) {
        return userRepository.save(User.newUser(name));
    }

    public void delete(UUID id) {
        Optional.ofNullable(userRepository.find(id)).ifPresent(userRepository::delete);
    }
}

File: src/main/java/com/github/miyashy/karatesample/service/UserService.kt
package com.github.miyashy.karatesample.service

import com.github.miyashy.karatesample.domain.User
import com.github.miyashy.karatesample.domain.User.Companion.newUser
import com.github.miyashy.karatesample.domain.repository.UserRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserService(val userRepository: UserRepository) {
    fun add(name: String) : User {
        return userRepository.save(newUser(name))
    }

    fun delete(id: UUID) {
        userRepository.find(id)?.let(userRepository::delete)
    }
}